__integrations__/__tests__/activatePubKey.test.ts

```diff
@@ -80,13 +98,291 @@ beforeEach(() => {
+    const body = await response.json();
+    expect(body).toMatchObject({
+      status: 400,
+      title: "Invalid AssertionRef"
+    });
+  });
+
+  it("should fail when an invalid payload is passed to the endpoint", async () => {
+    const response = await fetchActivatePubKey(
+      aValidSha256AssertionRef,
+      { ...validActivatePubKeyPayload, fiscal_code: "anInvalidFiscalCode" },
+      baseUrl,
+      (myFetch as unknown) as typeof fetch
+    );
+
+    expect(response.status).toEqual(400);
+    const body = await response.json();
+    expect(body).toMatchObject({
+      status: 400,
+      title: "Invalid ActivatePubKeyPayload"
+    });
+  });
+});
+
+describe("activatePubKey |> Failures", () => {
+  it("should return 404 when document cannot be found in cosmos", async () => {
+    const response = await fetchActivatePubKey(
+      aValidSha256AssertionRef,
+      validActivatePubKeyPayload,
+      baseUrl,
+      (myFetch as unknown) as typeof fetch
+    );
+
+    expect(response.status).toEqual(404);
+    const body = await response.json();
+    expect(body).toMatchObject({
+      status: 404,
+      title: "NotFound"
+    });
+  });
+
+  it("should return 500 when the retrieved pop document has status != PENDING", async () => {
+    const randomJwk = await generateJwkForTest();
+    const randomAssertionRef = await generateAssertionRefForTest(randomJwk);
+    const randomAssertionFileName = `${aFiscalCode}-${randomAssertionRef}` as AssertionFileName;
+
+    const randomNewPopDocument: NewLolliPopPubKeys = {
+      pubKey: toEncodedJwk(randomJwk),
+      ttl: TTL_VALUE_FOR_RESERVATION,
+      assertionRef: randomAssertionRef,
+      status: PubKeyStatusEnum.REVOKED,
+      assertionFileName: randomAssertionFileName,
+      assertionType: AssertionTypeEnum.SAML,
+      fiscalCode: aFiscalCode,
+      expiredAt: expires
+    };
+    const randomActivatePubKeyPayload: ActivatePubKeyPayload = {
+      fiscal_code: aFiscalCode,
+      assertion: "aValidAssertion" as NonEmptyString,
+      assertion_type: AssertionTypeEnum.SAML,
+      expires_at: expires
+    };
+
+    const res = await lolliPOPKeysModel.create(randomNewPopDocument)();
+
+    expect(res._tag).toEqual("Right");
+
+    const response = await fetchActivatePubKey(
+      randomAssertionRef,
+      randomActivatePubKeyPayload,
+      baseUrl,
+      (myFetch as unknown) as typeof fetch
+    );
+
+    expect(response.status).toEqual(500);
+    const body = await response.json();
+    expect(body).toMatchObject({
+      status: 500,
+      title: "Internal server error",
+      detail: "Unexpected status on pop document during activation: REVOKED"
+    });
+  });
+});
+
 describe("activatePubKey |> Success Results", () => {
-  it("dummy", () => {
-    expect(true).toBe(true);
+  it("should succeed when valid payload is passed to the endpoint AND when algo != master", async () => {
+    // TODO: replace with insert call (POST /api/v1/pubKeys)
```

Comment:
TODO here

Reply:
yes, you're right. i totally forgot it, my bad. I'll add it